src/components/button-toggle/button-toggle.ts

```diff
@@ -0,0 +1,327 @@
+})
+export class MdButtonToggleGroupMultiple {
+  /** Disables all toggles in the group. */
+  private _disabled: boolean = null;
+
+  /** Child button toggles. */
+  @ContentChildren(forwardRef(() => MdButtonToggle))
+  private _buttonToggles: QueryList<MdButtonToggle> = null;
+
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+
+  set disabled(value) {
+    this._disabled = (value != null && value !== false) ? true : null;
+  }
+}
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-button-toggle',
+  templateUrl: 'button-toggle.html',
+  styleUrls: ['button-toggle.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class MdButtonToggle implements OnInit {
+  /** Whether or not this button toggle is checked. */
+  private _checked: boolean = false;
+
+  /** Type of the button toggle. Either 'radio' or 'checkbox'. */
+  private _type: string = null;
+
+  /** The unique ID for this button toggle. */
+  @HostBinding('id')
+  @Input()
+  id: string;
+
+  /** HTML's 'name' attribute used to group radios for unique selection. */
+  @Input()
+  name: string;
+
+  /** Whether or not this button toggle is disabled. */
+  private _disabled: boolean = null;
+
+  /** Value assigned to this button toggle. */
+  private _value: any = null;
+
+  /** The parent button toggle group (exclusive selection). Optional. */
+  buttonToggleGroup: MdButtonToggleGroup;
+
+  /** The parent button toggle group (multiple selection). Optional. */
+  buttonToggleGroupMultiple: MdButtonToggleGroupMultiple;
+
+  /** Event emitted when the group value changes. */
+  @Output()
+  change: EventEmitter<MdButtonToggleChange> = new EventEmitter<MdButtonToggleChange>();
+
+  constructor(@Optional() toggleGroup: MdButtonToggleGroup,
+              @Optional() toggleGroupMultiple: MdButtonToggleGroupMultiple,
+              public buttonToggleDispatcher: MdUniqueSelectionDispatcher) {
+    this.buttonToggleGroup = toggleGroup;
+
+    this.buttonToggleGroupMultiple = toggleGroupMultiple;
+
+    buttonToggleDispatcher.listen((id: string, name: string) => {
+      if (id != this.id && name == this.name) {
+        this.checked = false;
+      }
+    });
+  }
+
+  ngOnInit() {
+    if (this.id == null) {
+      this.id = `md-button-toggle-${_uniqueIdCounter++}`;
+    }
+
+    if (this.buttonToggleGroup && this._value == this.buttonToggleGroup.value) {
+      this._checked = true;
+    }
+
+    // Even if there is no group at all, treat the button toggle as a checkbox so it can be
+    // toggled on or off.
+    if (this.buttonToggleGroup) {
+      this._type = 'radio';
+      this.name = this.buttonToggleGroup.name;
+    } else {
+      this._type = 'checkbox';
+    }
+  }
+
+  get inputId(): string {
+    return `${this.id}-input`;
+  }
+
+  @HostBinding('class.md-button-toggle-checked')
+  @Input()
+  get checked(): boolean {
+    return this._checked;
+  }
+
+  set checked(value: boolean) {
+    if (value && (this.buttonToggleGroup)) {
```

Comment:
I would add a getter like `isSingleSelection` and use that whenever you need to check if this is a radio-flavor toggle, since it's more semantically readable than checking the presence of the toggle group.